gwpy/io/datafind.py

```diff
@@ -316,40 +120,55 @@ def on_tape(*files):
         otherwise `False`
     """
     for path in files:
+        stat = os.stat(path)
         try:
-            if os.stat(path).st_blocks == 0:
+            if stat.st_blocks == 0:
                 return True
         except AttributeError:  # windows doesn't have st_blocks
             return False
     return False
 
 
-def _choose_connection(**datafind_kw):
-    if os.getenv('LIGO_DATAFIND_SERVER') or datafind_kw.get('host'):
-        from gwdatafind import connect
-        return connect(**datafind_kw)
-    if os.getenv('VIRGODATA'):
-        return FflConnection()
-    raise RuntimeError("unknown datafind configuration, cannot discover data")
+def _gwdatafind_module(**datafind_kw):
+    """Return the appropriate GWDataFind-like API based on the environment
+
+    This allows switching to the hacky `gwpy.io.ffldatafind` replacement
+    module to enable a GWDataFind-like interface for direct FFL data
+    discovery at Virgo.
+    """
+    # GWDataFind
+    if (
+        os.getenv('GWDATAFIND_SERVER')
```

Comment:
I can't comment on line 28 for some reason (??), but the docstring should probably be updated to note that the `GWDATAFIND_SERVER` environment variable is also an option, in addition to `LIGO_DATAFIND_SERVER`.

Reply:
(You can only comment on lines that have been impacted by the changes in the PR, I think).